lib/gitlib.sh

```diff
@@ -39,16 +33,48 @@ BRANCH_REGEX="master|release-([0-9]{1,})\.([0-9]{1,})(\.([0-9]{1,}))*$"
 # release - 1=Major, 2=Minor, 3=Patch, 4=-(alpha|beta|rc), 5=rev
 # dotzero - 1=Major, 2=Minor
 # build - 1=build number, 2=sha1
-declare -A VER_REGEX=([release]="v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-[a-zA-Z0-9]+)*\.*(0|[1-9][0-9]*)?"
-                      [dotzero]="v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.0$"
-                      [build]="([0-9]{1,})\+([0-9a-f]{5,40})"
-                     )
+# Define what a "digit" looks like
+# v is optional (Istio)
+declare -A VER_REGEX=([digit]="(0|[1-9][0-9]*)")
+VER_REGEX+=([x.y]="v?${VER_REGEX[digit]}\.${VER_REGEX[digit]}")
+VER_REGEX+=([release]="${VER_REGEX[x.y]}\.${VER_REGEX[digit]}(-[a-zA-Z0-9]+)*\.*${VER_REGEX[digit]}?"
+            [dotzero]="${VER_REGEX[x.y]}\.0$"
+            [build]="([0-9]{1,})\+([0-9a-f]{5,40})"
+           )
 
 ###############################################################################
 # FUNCTIONS
 ###############################################################################
 
 ###############################################################################
+# Set API roots and git URLs
+# @param project_repo - defaults to kubernetes/kubernetes
+# @optparam --use-remote - instruct to try to get a remote from git clone
+# Sets GLOBALS for use throughout
+gitlib::set_api_roots () {
+  local project_repo=$1
+  local remote
+
+  if [[ $project_repo == "--use-remote" ]]; then
+    if remote=$(git config --get remote.origin.url) && \
+       [[ $remote =~ .*github.com.(.*) ]]; then
+      project_repo="${BASH_REMATCH[1]}"
+    fi
+  fi
+
+  PROJECT_REPO=${project_repo:-"kubernetes/kubernetes"}
+  PROJECT=${PROJECT_REPO%/*}
+
+  K8S_GITHUB_API_ROOT="https://api.github.com/repos"
```

Comment:
should these vars still have `K8S_` as a prefix?

Reply:
The hope is that this is a very short term fix soon to be folded into #434 and trying to reduce the amount of unnecessary changes.  As well the fact that this is still under the K8S org.